game_eggs/yuzu-emu/yuzu-emu.json

```diff
@@ -0,0 +1,102 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2023-02-19T12:36:09+01:00",
+    "name": "YuzuEmu",
+    "author": "Z3r0byte",
+    "description": "Egg for hosting dedicated Yuzu multiplayer rooms.",
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:debian": "ghcr.io\/parkervcp\/yolks:debian"
+    },
+    "file_denylist": [],
+    "startup": ".\/yuzu-room --room-name \"{{ROOM_NAME}}\"  --room-description \"{{ROOM_DESCRIPTION}}\"  --preferred-game \"{{PREFERRED_GAME}}\"  --preferred-game-id {{PREFERRED_GAME_ID}}  --port {{SERVER_PORT}}  --max_members {{MAX_MEMBERS}} --password \"{{PASSWORD}}\" --token \"{{YUZU_TOKEN}}\" --web-api-url \"https:\/\/api.yuzu-emu.org\/\"",
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"Room is open\"\r\n}",
+        "logs": "{}",
+        "stop": "Q"
+    },
+    "scripts": {
+        "installation": {
+            "script": "# Install build tools\r\necho \"deb http:\/\/deb.debian.org\/debian bullseye-backports main\" >> \/etc\/apt\/sources.list && \\\r\n    apt-get update && apt-get -y full-upgrade && \\\r\n    apt-get install -y build-essential wget git ccache ninja-build libssl-dev pkg-config libarchive-tools \\\r\n    cmake\/bullseye-backports cmake-data\/bullseye-backports \r\n\r\n\r\n# Shallow clone yuzu repository, since we delete the local copy afterwards\r\ngit clone --depth 1 -j4 --recursive --shallow-submodules https:\/\/github.com\/yuzu-emu\/yuzu-mainline.git \/mnt\/server\/yuzu-mainline\r\n\r\n# Download build tools and build\r\ngit clone --depth 1 https:\/\/github.com\/yuzu-emu\/yuzu-multiplayer-dedicated.git \/mnt\/server\/build-files\r\ncd \/mnt\/server\/yuzu-mainline && \/mnt\/server\/build-files\/.ci\/build.sh\r\n\r\n# Copy server binary to container root\r\ncp \/mnt\/server\/yuzu-mainline\/build\/bin\/yuzu-room \/mnt\/server\/yuzu-room\r\n\r\n# Delete git repo's to save disk space.\r\nrm -rf \/mnt\/server\/yuzu-mainline && rm -rf \/mnt\/server\/build-files",
```

Comment:
install script first line always must be: `#!/bin/bash`

And wy are you even buildign this and importing a backport repo? I made a egg that uses wine in that PR.
We normaly don't build anything in a install container only if it is 100% needed and it runs fine with wine

Reply:
The backport repo was necessary for the build to succeed, I followed the same procedure the yuzu team uses in their docker container for the server: https://github.com/yuzu-emu/yuzu-multiplayer-dedicated/blob/master/Dockerfile

As for why I compiled it, instead of using wine: I did indeed see your wine-based egg, but since a pull request was never opened for it, I assumed it was either not ready/non-optimal. Plus, I also kinda like native solutions more (the resources needed to run the server are really, really low, I haven't seen it use more than 5MiB of RAM yet).

Feel free to instead merge your solution though, I just created this PR because no-one else had done so before.

Reply:
The wine version indeed uses more resources. But the binary was prebuild. We like to avoid compiling in the install container. To speed it up as we like to keep install time under like 5 or 10 minutes. There are exceptions like pockitme on arm64.

Reply:
I understand. In my testing installing takes a little under 4 minutes on a ryzen 3600 server, so I expect most servers to be able to handle it under that 10 minute mark.

Reply:
not everyone has that fast hardware. I think something like a xeon then it will take a wile. but I will try it

Reply:
I see that there is not another option, while I don't realy like that is compiling it, it is the only way so fine for me